File: rails-api/app/models/olympian.rb
class Olympian < ApplicationRecord
  before_validation :convert_NA_to_nil

  def convert_NA_to_nil
    self.age = age == 0 ? nil : age
    self.height = height == 0 ? nil : height
    self.weight = weight == 0 ? nil : weight
  end

  def self.index
    data = select(:id, :name, :sex, :age, :height, :weight)
    return { olympians: data }
  end
end

File: go/src/golangApi/models/olympian.go
package olympian

import "database/sql"

type OlympianSummary struct {
	ID         int
	Name       string
  Sex        string
  Age        int
	Height     sql.NullString
	Weight     sql.NullString
}

type Olympians struct {
	Olympians []OlympianSummary
}
